### ISSUE TITLE
Feat: ability to set focus on InputText

### ISSUE THREAD

**Issue by DreadIsBack** [primefaces/primevue#2397]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => Search github for a similar issue or PR before submitting
[x] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primevue/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=110
```

**Description**
Please add ability to set focus on `InputText`. Thank you.
For example it will be used like: `this.$refs.inputText.focus();`

**Comment by tugcekucukoglu** [comment]
You could use `autofocus ` attribute.

**Comment by ashokgelal** [comment]
`autofocus` is only good for when a page is loaded or a component is mounted for the first time. There are other use cases for being able to programmatically set focus:
* hiding/unhiding of a UI
* focusing on actual input box in case there was an error coming from the backend etc.

Both Vue2 and Vue3 offer a way to expose a method from within a component and a `focus()` method could be exposed from the InputText/Password components.

**Comment by stripedbug** [comment]
Just use component ref.

```
<InputText   ref="inputarea"  />
this.$refs.inputarea.$el.focus()
```

**Comment by ltlBeBoy** [comment]
@DreadIsBack Simply assign your component an ID and use:
`this.$el.querySelector("#yourId").focus();`

**Comment by gonzalo-meto** [comment]
I think It could be better if it exposes the actual HTML input that is representing

**Comment by reksc** [comment]
For vue 3, assuming that `input = ref()`, you can do
```
input.value.$el.focus();
```

**Comment by ashokgelal** [comment]
@reksc @ltlBeBoy yes, there is a way to do that by digging into the the html ref, we know it. But what we are asking is having a nice "API" rather than having to get the underlying input element.

**Comment by AEROGU** [comment]
> For vue 3, assuming that `input = ref()`, you can do
> 
> ```
> input.value.$el.focus();
> ```

Thanks @reksc , that was the best and most elegant way to do it.

**Comment by imad-mn** [comment]
> For vue 3, assuming that `input = ref()`, you can do
> 
> ```
> input.value.$el.focus();
> ```

This didn´t work for me. I could not make it working

**Comment by vcmenezes** [comment]
> > For vue 3, assuming that `input = ref()`, you can do
> > ```
> > input.value.$el.focus();
> > ```
> 
> This didn´t work for me. I could not make it working

@imad-mn Try inside nextTick:
```javascript
nextTick(() => {
   input.value.$el.focus();
});
```

**Comment by satezmedia** [comment]
@imad-mn Try to use 
`await nextTick()`
That solved my use case

**Comment by Hulkmaster** [comment]
eh, still closed?

i just want to double everybody else said: would be nice to have autofocus on ipnut which would focus on itself upon mounting

**Comment by jumika** [comment]
> For vue 3, assuming that `input = ref()`, you can do
> 
> ```
> input.value.$el.focus();
> ```

We are using typescript so I ended up with this beauty:
```
(passwordInputRef.value as null | { $el?: HTMLInputElement })?.$el?.focus();
```
:D

**Comment by julisch94** [comment]
I'm not sure why this is completed. When I use `autofocus` it only focusses on the first mount. How am I supposed to do this. I could use 
```ts
const onAddShowClicked = () => {
  nextTick(() => title.value?.focus())
}

<input ref="title" />
```
but that's not PrimeVue.

https://github.com/primefaces/primevue/assets/22644876/63e2b36b-8d40-445d-aea7-28dbada2791f

**Comment by crcastle** [comment]
For any still struggling with this one, here's a fairly clean way to do it. It ensures the component is actually mounted before trying to set focus on the input element. I believe this should get around the need to use `nextTick()`. This also gets around the problem of the `autofocus` attribute only working when the page is first loaded. And also doesn't require an extra `ref()`.

```vue
<script setup lang="ts">
import { ref } from 'vue';

const value = ref('');

const textFocus = (el: any) => {
  el.el?.focus();

  // I needed to use this with the InputOtp component and used the following instead
  // el.el?.firstChildElement.focus()
}
</script>

<template>
  <InputText type="text" v-model="value" @vue:mounted="textFocus" />
</template
```

Here's a live example with more details: https://stackblitz.com/edit/edghca?file=src%2FApp.vue

**Comment by mmghv** [comment]
@crcastle Thank you, this is the same but inlined as it's simple enough :

```vue
<InputText v-model="value" @vue:mounted="$event.el?.focus()" />
```

**Comment by kresohr** [comment]
> For vue 3, assuming that `input = ref()`, you can do
> 
> ```
> input.value.$el.focus();
> ```

God Bless.
Still working after 3 years lol

**Comment by rzschoch** [comment]
I implemented it this way in a TypeScript project:

```typescript
const usernameOrEmailInput = useTemplateRef<{
  $el: HTMLInputElement;
}>('usernameOrEmailInput');

onStartTyping(() => {
  if (usernameOrEmailInput.value && usernameOrEmailInput.value.$el !== document.activeElement) {
    usernameOrEmailInput.value.$el.focus();
  }
});
```

**Comment by attilakreiner** [comment]
```typescript
const inputRef = useTemplateRef<ComponentPublicInstance<{ $el: HTMLElement }>>('input')

function focusInput() {
  nextTick(() => {
    inputRef.value?.$el.focus()
  })
}
```
